src/rpc/rawtransaction.cpp

```diff
@@ -1470,6 +1470,139 @@ UniValue sendrawtransaction(const UniValue &params, bool fHelp)
+            entry_metadata.pushKV(it->first, it->second);
+        }
+        entry.pushKV("metadata", entry_metadata);
+        UniValue entry_errors(UniValue::VARR);
+        for (auto error : data.errors)
+        {
+            entry_errors.push_back(error);
+        }
+        entry.pushKV("errors", entry_errors);
+        uv_vdata.push_back(entry);
+    }
+    result.pushKV("inputs", uv_vdata);
+}
+
+UniValue validaterawtransaction(const UniValue &params, bool fHelp)
+{
+    if (fHelp || params.size() < 1 || params.size() > 3)
+    {
+        throw std::runtime_error(
+            "validaterawtransaction \"hexstring\" ( allowhighfees, allownonstandard )\n"
+            "\nValidates raw transaction (serialized, hex-encoded) to local node without broadcasting it.\n"
+            "\nAlso see createrawtransaction and signrawtransaction calls.\n"
+            "\nArguments:\n"
+            "1. \"hexstring\"    (string, required) The hex string of the raw transaction)\n"
+            "2. allowhighfees    (boolean, optional, default=false) Allow high fees\n"
+            "3. allownonstandard (string 'standard', 'nonstandard', 'default', optional, default='default')\n"
+            "                    Force standard or nonstandard transaction check\n"
+            "\nResult:\n"
+            "\"hex\"             (string) The transaction hash in hex\n"
+            "\nExamples:\n"
+            "\nCreate a transaction\n" +
+            HelpExampleCli("createrawtransaction",
+                "\"[{\\\"txid\\\" : \\\"mytxid\\\",\\\"vout\\\":0}]\" \"{\\\"myaddress\\\":0.01}\"") +
+            "Sign the transaction, and get back the hex\n" + HelpExampleCli("signrawtransaction", "\"myhex\"") +
+            "\nSend the transaction (signed hex)\n" + HelpExampleCli("sendrawtransaction", "\"signedhex\"") +
+            "\nAs a json rpc call\n" + HelpExampleRpc("validaterawtransaction", "\"signedhex\""));
+    }
+
+    RPCTypeCheck(params, boost::assign::list_of(UniValue::VSTR)(UniValue::VBOOL)(UniValue::VSTR));
+
+    // parse hex string from parameter
+    CTransaction tx;
+    if (!DecodeHexTx(tx, params[0].get_str()))
+        throw JSONRPCError(RPC_DESERIALIZATION_ERROR, "TX decode failed");
+    CTransactionRef ptx(MakeTransactionRef(std::move(tx)));
+    const uint256 &hashTx = ptx->GetHash();
+
+    bool fOverrideFees = false;
+    TransactionClass txClass = TransactionClass::DEFAULT;
+
+    // 2nd parameter allows high fees
+    if (params.size() > 1)
+    {
+        fOverrideFees = params[1].get_bool();
+    }
+    // 3rd parameter must be the transaction class
+    if (params.size() > 2)
+    {
+        txClass = ParseTransactionClass(params[2].get_str());
+        if (txClass == TransactionClass::INVALID)
+            throw JSONRPCError(RPC_INVALID_PARAMETER, "Invalid transaction class");
+    }
+
+    CCoinsViewCache &view = *pcoinsTip;
+    bool fHaveChain = false;
+    {
+        for (size_t o = 0; !fHaveChain && o < tx.vout.size(); o++)
```

Comment:
once upon a time we never used o as a variable name because it looks too much like 0.  Perhaps that's no longer relevant for many fonts, but at the same time, let's keep to that standard so it doesn't bug external reviewers.

Reply:
fixed